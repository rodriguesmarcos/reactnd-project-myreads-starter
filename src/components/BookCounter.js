import React from 'react';
import PropTypes from 'prop-types';

const BookCounter = ({total, showing = total}) => {

  if ( total === showing )
    return <span className="bookshelf-counter">{total}</span>;

  if ( total > showing )
    return <span className="bookshelf-counter">{`${showing} of ${total}`}</span>;

};

BookCounter.propTypes = {
  total: PropTypes.number.isRequired,
  showing: PropTypes.number,
}

export default BookCounter;
