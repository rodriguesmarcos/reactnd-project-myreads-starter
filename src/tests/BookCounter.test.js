import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import BookCounter from '../components/BookCounter.js';

describe('<BookCounter />', () => {

  const minProps = {
    total: 10,
  }

  test('renders without exploding', () => {
    const wrapper = shallow(<BookCounter {...minProps}/>);
    expect(wrapper.length).toBe(1);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('should show total', () => {
    const wrapper = shallow(<BookCounter {...minProps} />);
    expect(wrapper.text()).toBe(`${minProps.total}`);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('should show "s of t"', () => {
    const showing = 5;
    const wrapper = shallow(<BookCounter {...minProps} showing={showing} />);
    expect(wrapper.text()).toBe(`${showing} of ${minProps.total}`);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

});
