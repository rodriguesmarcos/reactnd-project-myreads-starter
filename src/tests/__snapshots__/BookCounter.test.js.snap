exports[`<BookCounter /> renders without exploding 1`] = `
<span
  className="bookshelf-counter">
  10
</span>
`;

exports[`<BookCounter /> should show "s of t" 1`] = `
<span
  className="bookshelf-counter">
  5 of 10
</span>
`;

exports[`<BookCounter /> should show total 1`] = `
<span
  className="bookshelf-counter">
  10
</span>
`;
